File: benchmark/results/python-multilang/run-003/b-baseline/output/metaengine_demo/shipping/services/carrier_gateway.py
"""CarrierGateway for the shipping domain."""
from typing import Optional

from metaengine_demo.shipping.aggregates.shipment import Shipment


class CarrierGateway:
    """CarrierGateway external integration service."""

    def create(self, input: Optional[Shipment]) -> Shipment:
        """Submit a new shipment to a carrier."""
        raise NotImplementedError("not implemented")

    def findById(self, id: str) -> Optional[Shipment]:
        """Lookup a shipment via the carrier gateway."""
        raise NotImplementedError("not implemented")

    def list(self, limit: float) -> list[Shipment]:
        """List carrier-tracked shipments."""
        raise NotImplementedError("not implemented")

    def delete(self, id: str) -> None:
        """Cancel a shipment with the carrier."""
        raise NotImplementedError("not implemented")

File: benchmark/results/ts-script-sonnet/run-003/b-baseline/output/src/domain/shipping/services/CarrierGateway.ts
import { Shipment } from '../aggregates/Shipment';

/** Gateway for communicating with external carrier APIs. */
export class CarrierGateway {
  create(input: Partial<Shipment>): Shipment {
    void input;
    throw new Error('not implemented');
  }

  findById(id: string): Shipment | null {
    void id;
    throw new Error('not implemented');
  }

  list(limit: number): Shipment[] {
    void limit;
    throw new Error('not implemented');
  }

  delete(id: string): void {
    void id;
    throw new Error('not implemented');
  }
}
